### ISSUE TITLE
failing to interpret an .spc file

### ISSUE THREAD

**Issue by jstoner** [rohanisaac/spc#4]
I am trying to interpret a file with your code, which I can send to you (apparently Github won't allow me to attach .spc files). It was produced by a spectrophotometer from Shimadzu, their UV1601PC- UV Visible Spectrophotometer. Apparently the file produced has a version ('fversn') of 0xfc. This library doesn't do anything with that version. When I try to use your convert tool it chokes looking for 'pr_xlabel,' which doesn't get set with a file with an unknown version.

Do you have any interest in supporting the output of that spectrophotometer? I'm fairly deep with Python, but the people in my lab know more about the equipment than I do.

**Comment by rohanisaac** [comment]
Yeah, I'd love to support more instruments. You can attach the file if you compress it first (Github allows .zip files). I'll take a look at and see if I can update the library.

**Comment by rohanisaac** [comment]
I made the requirement of setting pr_xlabel optional for data output, but I haven't tested it. If that was the only issue, it might be fixed.

**Comment by ePostler** [comment]
I tried to interpret my SPC file with your code. Unfortunately, my file format (0x28) is not yet supported. The spectrum is from a FEI TECHNAI F20.
[FEI_TECNAI_F20.zip](https://github.com/rohanisaac/spc/files/214984/FEI_TECNAI_F20.zip)

**Comment by rohanisaac** [comment]
I'll take a look and see what I can do, unfortunately it is hard to extract data reliably from some of these more obscure binary formats without some kind of documentation on the specific format they use.

**Comment by sfadschm** [comment]
To whoever still needs to convert Shimadzu SPC-files:
Those files are not really binary files. They in fact are OLE storages which can be treated as archives. E.g., these files can be extracted by 7zip. Inside you will find all Data and Info in Hexadecimal notation.
The data can fairly easily be extracted e.g. by a VBScript (which is what I'm using) or Python.

**Comment by rohanisaac** [comment]
Oh that's really interesting. Thanks for the information, that is quite useful.

**Comment by sfadschm** [comment]
Welcome, if you need any example spc files or other Info, just let me know.
